### ISSUE TITLE
How to self define metric instead of using one of the pre-defined metrics

### ISSUE THREAD

**Issue by CFZhai** [shenweichen/DeepCTR#536]
How to self define metric instead of using one of the pre-defined metrics

**Comment by CFZhai** [comment]
or how to add a sklearn metric into 

    def _get_metrics(self, metrics, set_eps=False):
        metrics_ = {}
        if metrics:
            for metric in metrics:
                if metric == "binary_crossentropy" or metric == "logloss":
                    if set_eps:
                        metrics_[metric] = self._log_loss
                    else:
                        metrics_[metric] = log_loss
                if metric == "auc":
                    metrics_[metric] = roc_auc_score
                if metric == "mse":
                    metrics_[metric] = mean_squared_error
                if metric == "accuracy" or metric == "acc":
                    metrics_[metric] = self._accuracy_score
                self.metrics_names.append(metric)
        return metrics_

**Comment by shenweichen** [comment]
For Keras based DeepCTR models, custom metrics can be passed directly to `model.compile`, for example `metrics=[my_metric]` or a `tf.keras.metrics.Metric` instance.

sklearn metrics usually operate on full numpy arrays after prediction, so use them in an evaluation callback or call them on `model.predict` output. The old `_get_metrics` helper only supports built-in names.

Closing as answered.
